packages/flutter/lib/src/services/predictive_back_event.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// Enum representing the edge from which a swipe starts in a back gesture.
+///
+/// This is used in [PredictiveBackEvent] to indicate the starting edge of the
+/// swipe gesture.
+enum SwipeEdge {
+  /// Indicates that the swipe gesture starts from the left edge of the screen.
+  left,
+
+  /// Indicates that the swipe gesture starts from the right edge of the screen.
+  right,
+}
+
+/// Object used to report back gesture progress in Android.
+///
+/// Holds information about the touch event, swipe direction and the animation
+/// progress that predictive back animations should seek to.
+@immutable
+class PredictiveBackEvent {
+  /// Creates a new [PredictiveBackEvent] instance.
+  const PredictiveBackEvent._({
+    required this.x,
+    required this.y,
+    required this.progress,
+    required this.swipeEdge,
+  }) : assert(progress >= 0.0 && progress <= 1.0);
+
+  /// Creates an [PredictiveBackEvent] from a Map, typically used when converting
+  /// data received from a platform channel.
+  factory PredictiveBackEvent.fromJSON(Map<String, dynamic> json) {
+    return PredictiveBackEvent._(
+      x: (json['x'] as num?)?.toDouble(),
+      y: (json['y'] as num?)?.toDouble(),
+      progress: (json['progress'] as num).toDouble(),
+      swipeEdge: SwipeEdge.values[json['swipeEdge'] as int],
+    );
+  }
+
+  /// The global X location of the touch point, or `null` if the event is from a
+  /// button press.
+  final double? x;
+
+  /// The global Y location of the touch point, or `null` if the event is from a
+  /// button press.
+  final double? y;
+
+  /// Returns a value between 0 and 1 representing how far along the back
```

Comment:
nit: should we use double literals in this doc comment as well? Looking through our repo I see a few instances of double literals in doc comments.

Reply:
I say let's do it :+1: